src/views/Project/index.jsx

```diff
@@ -0,0 +1,168 @@
+import { Component } from 'react';
+import Grid from '@material-ui/core/Grid';
+import Typography from '@material-ui/core/Typography';
+import ReactMarkdown from 'react-markdown';
+import { ApolloConsumer } from 'react-apollo';
+import _ from 'lodash';
+import gql from 'graphql-tag';
+import projects from '../../data/loader';
+import BugsTable from '../../components/BugsTable';
+import Issues from './issues.graphql';
+
+class Project extends Component {
+  constructor(props) {
+    super(props);
+    const fileName = props.match.params.projectKey;
+    const projectInfo = projects[fileName];
+    // dictionary with repo as key and tag as value
+    const tmp = projectInfo.repositories
+      ? projectInfo.repositories.reduce(
+          (prev, cur) => ({ ...prev, ...cur }),
+          {}
+        )
+      : {};
+    // initial query search for all repo
+    const repoQuerySearch = ['state:open'];
+    // a dictionary of tag (string) and list of repository in that tag
+    const tagRepoDictionary = Object.keys(tmp).reduce((prev, key) => {
+      const curr = [
+        ...(prev[tmp[key]] || [...repoQuerySearch, `label:${tmp[key]}`]),
+        `repo:${key}`,
+      ];
+
+      return {
+        ...prev,
+        [tmp[key]]: curr,
+      };
+    }, {});
+    const tagRepoList = Object.entries(tagRepoDictionary).map(
+      ([key, value]) => ({
+        label: key,
+        query: value.join(' '),
+      })
+    );
+
+    this.state = {
+      projectInfo,
+      tagRepoList,
+      error: false,
+      loading: true,
+      data: [],
+    };
+  }
+  componentDidMount() {
+    const { tagRepoList } = this.state;
+
+    this.fetchGithubDataNext(tagRepoList);
+  }
+
+  fetchGithubDataNext(tagRepoList) {
+    const { client } = this.props;
+    let allQuery = '';
+
+    if (tagRepoList.length === 0) return;
+
+    tagRepoList.forEach((tag, idx) => {
+      const noCursorQuery = `_${idx}: search(first:100, type:ISSUE, query:"${
+        tag.query
+      }")\n
+      {
+      ...Issues
+      }\n`;
+
+      allQuery = allQuery.concat(noCursorQuery);
+    });
+
+    client
+      .query({ query: gql`{${allQuery}}\n${Issues}` })
```

Comment:
Could we use the decorator approach instead? For inspiration, you can look at a:

Simple query: https://github.com/taskcluster/taskcluster-web/blob/2998f6561a4de26908d4a2eba4aaca8443c15a53/src/views/Provisioners/ViewProvisioners/index.jsx#L13

Query with variables: https://github.com/taskcluster/taskcluster-web/blob/2998f6561a4de26908d4a2eba4aaca8443c15a53/src/views/Provisioners/ViewWorkers/index.jsx#L19-L30